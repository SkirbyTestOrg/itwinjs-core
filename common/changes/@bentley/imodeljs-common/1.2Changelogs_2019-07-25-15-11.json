{
  "changes": [
    {
      "comment": "",
      "packageName": "@bentley/imodeljs-common",
      "type": "none"
    }
  ],
  "packageName": "@bentley/imodeljs-common",
  "email": "31107829+calebmshafer@users.noreply.github.com"
}