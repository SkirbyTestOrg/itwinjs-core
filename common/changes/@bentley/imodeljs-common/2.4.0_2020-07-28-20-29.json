{
  "changes": [
    {
      "comment": "",
      "type": "none",
      "packageName": "@bentley/imodeljs-common"
    }
  ],
  "packageName": "@bentley/imodeljs-common",
  "email": "Paul.Connelly@bentley.com"
}