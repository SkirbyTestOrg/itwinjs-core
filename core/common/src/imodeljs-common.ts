/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
export * from "./Code";
export * from "./ColorDef";
export * from "./ECSqlTypes";
export * from "./ElementProps";
export * from "./EntityProps";
export * from "./FeatureGates";
export * from "./FeatureIndex";
export * from "./Frustum";
export * from "./Fonts";
export * from "./GeoCoordinateServices";
export * from "./RpcInterface";
export * from "./RpcManager";
export * from "./TextureProps";
export * from "./Image";
export * from "./IModel";
export * from "./IModelError";
export * from "./IModelVersion";
export * from "./Lighting";
export * from "./CommonLoggerCategory";
export * from "./MaterialProps";
export * from "./ModelProps";
export * from "./OctEncodedNormal";
export * from "./QPoint";
export * from "./SubCategoryAppearance";
export * from "./Snapping";
export * from "./TileProps";
export * from "./SpatialClassificationProps";
export * from "./Thumbnail";
export * from "./ViewProps";
export * from "./Render";
export * from "./Paging";
export * from "./RenderSchedule";
export * from "./ChangedElements";
export * from "./domains/FunctionalElementProps";
export * from "./domains/GenericElementProps";
export * from "./domains/LinearReferencingElementProps";
export * from "./geometry/AreaPattern";
export * from "./geometry/Cartographic";
export * from "./geometry/GeometryStream";
export * from "./geometry/LineStyle";
export * from "./geometry/Placement";
export * from "./geometry/TextString";
export * from "./rpc/TestRpcManager";
export * from "./rpc/WipRpcInterface";
export * from "./rpc/DevToolsRpcInterface";
export * from "./rpc/core/RpcConstants";
export * from "./rpc/core/RpcConfiguration";
export * from "./rpc/core/RpcInvocation";
export * from "./rpc/core/RpcOperation";
export * from "./rpc/core/RpcProtocol";
export * from "./rpc/core/RpcRegistry";
export * from "./rpc/core/RpcRequest";
export * from "./rpc/core/RpcRequestContext";
export * from "./rpc/core/RpcControl";
export * from "./rpc/core/RpcMarshaling";
export * from "./rpc/core/RpcPendingQueue";
export * from "./rpc/electron/ElectronRpcManager";
export * from "./rpc/electron/ElectronRpcProtocol";
export * from "./rpc/electron/ElectronRpcRequest";
export * from "./rpc/mobile/MobileRpcManager";
export * from "./rpc/mobile/MobileRpcProtocol";
export * from "./rpc/mobile/MobileRpcRequest";
export * from "./rpc/web/BentleyCloudRpcManager";
export * from "./rpc/web/BentleyCloudRpcProtocol";
export * from "./rpc/web/OpenAPI";
export * from "./rpc/web/WebAppRpcProtocol";
export * from "./rpc/web/RpcMultipart";
export * from "./rpc/web/WebAppRpcRequest";
export * from "./rpc/mobile/MobileRpcManager";
export * from "./rpc/IModelReadRpcInterface";
export * from "./rpc/IModelTileRpcInterface";
export * from "./rpc/IModelWriteRpcInterface";
export * from "./rpc/SnapshotIModelRpcInterface";
export * from "./CloudStorage";
export * from "./CloudStorageTileCache";

// Set the version number so it can be found at runtime. BUILD_SEMVER is replaced at build time by the webpack DefinePlugin.
declare var BUILD_SEMVER: string;
if ((typeof (BUILD_SEMVER) !== "undefined") && (typeof window !== "undefined") && window) {
  if (!(window as any).iModelJsVersions)
    (window as any).iModelJsVersions = new Map<string, string>();
  (window as any).iModelJsVersions.set("imodeljs-common", BUILD_SEMVER);
}

/** @docs-package-description
 * The imodeljs-common package contains classes for working with iModels that can be used in both [frontend]($docs/learning/frontend/index.md) and [backend]($docs/learning/backend/index.md).
 */
/**
 * @docs-group-description WireFormats
 * Definitions of the "props" interfaces and types that define the [wire format]($docs/learning/wireformat.md) for communication between the frontend and backend
 */
/**
 * @docs-group-description Codes
 * Classes for working with [Codes]($docs/bis/intro/codes.md)
 */
/**
 * @docs-group-description Geometry
 * Classes for working with geometry.
 */
/**
 * @docs-group-description Views
 * Classes for working with views of models and elements.
 */
/**
 * @docs-group-description Rendering
 * Classes for rendering geometry in views.
 */
/**
 * @docs-group-description Symbology
 * Classes that affect the appearance of geometry in a view
 */
/**
 * @docs-group-description iModels
 * Classes for working with [iModels]($docs/learning/IModels.md) in both the frontend and backend
 */
/**
 * @docs-group-description RpcInterface
 * Classes for working with [RpcInterfaces]($docs/learning/RpcInterface.md).
 */
/**
 * @docs-group-description ECSQL
 * Classes for working with [ECSQL]($docs/learning/ECSQL.md), [Spatial Queries]($docs/learning/SpatialQueries.md), and [ECSQL Geometry Functions]($docs/learning/GeometrySqlFuncs.md)
 */
