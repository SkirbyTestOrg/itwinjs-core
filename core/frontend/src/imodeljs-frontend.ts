/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/

export * from "./tools/AccuDrawTool";
export * from "./tools/EditManipulator";
export * from "./tools/ElementSetTool";
export * from "./tools/EventController";
export * from "./tools/MeasureTool";
export * from "./tools/SelectTool";
export * from "./tools/ToolAdmin";
export * from "./tools/Tool";
export * from "./tools/ViewTool";
export * from "./tools/ClipViewTool";
export * from "./tools/PrimitiveTool";
export * from "./tools/IdleTool";
export * from "./tools/ToolAssistance";
export * from "./AccuDraw";
export * from "./AccuSnap";
export * from "./AuxCoordSys";
export * from "./DevTools";
export * from "./FrontendRequestContext";
export * from "./CategorySelectorState";
export * from "./ContextRealityModelState";
export * from "./SpatialClassification";
export * from "./DisplayStyleState";
export * from "./ElementLocateManager";
export * from "./EmphasizeElements";
export * from "./EntityState";
export * from "./FuzzySearch";
export * from "./GeoServices";
export * from "./HitDetail";
export * from "./IModelConnection";
export * from "./ImageUtil";
export * from "./FrontendLoggerCategory";
export * from "./Marker";
export * from "./ModelSelectorState";
export * from "./ModelState";
export * from "./NotificationManager";
export * from "./plugin/Plugin";
export * from "./plugin/PluginUiManager";
export * from "./RelativePosition";
export * from "./RenderScheduleState";
export * from "./SelectionSet";
export * from "./Sheet";
export * from "./Sprites";
export * from "./StandardView";
export * from "./SubCategoriesCache";
export * from "./TentativePoint";
export * from "./QuantityFormatter";
export * from "./ViewContext";
export * from "./ViewManager";
export * from "./Viewport";
export * from "./ViewState";
export * from "./IModelApp";
export * from "./NoRenderApp";
export * from "./tile/TileAdmin";
export * from "./tile/Tile";
export * from "./tile/TileTree";
export * from "./tile/WebMapTileTree";
export * from "./tile/BackgroundTerrainTileTree";
export * from "./tile/CesiumWorldTerrainTileTree";
export * from "./tile/BingElevation";
export * from "./render/FeatureSymbology";
export * from "./render/GraphicBuilder";
export * from "./render/MockRender";
export * from "./render/System";
export * from "./render/webgl/Target";
export * from "./RenderCompatibility";
export * from "./oidc/OidcBrowserClient";
export * from "./TerrainProvider";
export * from "./properties/Description";
export * from "./properties/EditorParams";
export * from "./properties/Record";
export * from "./properties/Value";
export * from "./properties/ToolSettingsValue";
export * from "./properties/PrimitiveTypes";
export * from "./properties/BaseQuantityDescription";
export * from "./properties/AngleDescription";
export * from "./properties/LengthDescription";
export * from "./SolarCalculate";

// Set the version number so it can be found at runtime. BUILD_SEMVER is replaced at build time by the webpack DefinePlugin.
declare var BUILD_SEMVER: string;
declare var BUILD_TYPE: string;
if ((typeof (BUILD_SEMVER) !== "undefined") && (typeof window !== "undefined") && window) {
  if (!(window as any).iModelJsVersions)
    (window as any).iModelJsVersions = new Map<string, string>();
  (window as any).iModelJsVersions.set("imodeljs-frontend", BUILD_SEMVER);
  (window as any).iModelJsVersions.set("buildType", BUILD_TYPE);
}

/** @docs-package-description
 * The ($frontend) package always runs in a web browser. It contains classes for [querying iModels and showing views]($docs/learning/frontend/index.md).
 */

/**
 * @docs-group-description IModelApp
 * Classes for configuring and administering an iModel.js application.
 * See [the learning articles]($docs/learning/frontend/index.md).
 */
/**
 * @docs-group-description IModelConnection
 * Classes for working with a connection to an [iModel briefcase]($docs/learning/IModels.md)
 */
/**
 * @docs-group-description ElementState
 * Classes for working with the *state* of Elements in the frontend.
 * See [the learning articles]($docs/learning/frontend/index.md).
 */
/**
 * @docs-group-description ModelState
 * Classes for working with the *state* of Models in the frontend.
 * See [the learning articles]($docs/learning/frontend/index.md).
 */
/**
 * @docs-group-description Tools
 * Classes for [working with Tools]($docs/learning/frontend/Tools.md)
 */
/**
 * @docs-group-description Views
 * Classes for [working with Views]($docs/learning/frontend/Views.md)
 */
/**
 * @docs-group-description LocatingElements
 * Classes for locating and snapping to elements in views.
 * See [the learning articles]($docs/learning/frontend/index.md).
 */
/**
 * @docs-group-description AccuDraw
 * AccuDraw provides helpful assistance for creating and modifying elements in a view.
 * See [the learning articles]($docs/learning/frontend/index.md).
 */
/**
 * @docs-group-description Notifications
 * Notifications provide feedback to the user of something of interest.
 * See [the learning articles]($docs/learning/frontend/index.md).
 */
/**
 * @docs-group-description Plugins
 * Classes for creating and managing runtime [Plugins]($docs/learning/frontend/Plugins.md)
 */
/**
 * @docs-group-description Properties
 * Classes for working with property records and descriptions.
 */
/**
 * @docs-group-description Rendering
 * Classes for rendering the contents of views.
 */
/**
 * @docs-group-description SelectionSet
 * Classes for working with the set of selected elements.
 * See [the learning articles]($docs/learning/frontend/index.md).
 */
