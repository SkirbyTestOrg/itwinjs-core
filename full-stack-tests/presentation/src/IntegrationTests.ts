/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
// tslint:disable:no-direct-imports
import "@bentley/presentation-frontend/lib/test/_helpers/MockFrontendEnvironment";
import * as chai from "chai";
import * as cpx from "cpx";
import * as fs from "fs";
import * as path from "path";
import sinonChai from "sinon-chai";
import { Logger, LogLevel } from "@bentley/bentleyjs-core";
import { IModelJsConfig } from "@bentley/config-loader/lib/IModelJsConfig";
import { IModelAppOptions, NoRenderApp } from "@bentley/imodeljs-frontend";
import { I18NOptions } from "@bentley/imodeljs-i18n";
import { TestUsers } from "@bentley/oidc-signin-tool/lib/TestUsers";
import { TestUtility } from "@bentley/oidc-signin-tool/lib/TestUtility";
import { Presentation as PresentationBackend, PresentationProps as PresentationBackendProps } from "@bentley/presentation-backend";
import { LoggingNamespaces, RequestPriority } from "@bentley/presentation-common";
import { PresentationManagerProps as PresentationFrontendProps } from "@bentley/presentation-frontend";
import { initialize as initializeTesting, PresentationTestingInitProps, terminate as terminateTesting } from "@bentley/presentation-testing";

chai.use(sinonChai);

IModelJsConfig.init(true);

const copyBentleyBackendAssets = (outputDir: string) => {
  const bentleyPackagesPath = "node_modules/@bentley";
  fs.readdirSync(bentleyPackagesPath).map((packageName) => {
    const packagePath = path.resolve(bentleyPackagesPath, packageName);
    return path.join(packagePath, "lib", "assets");
  }).filter((assetsPath) => {
    return fs.existsSync(assetsPath);
  }).forEach((src) => {
    cpx.copySync(`${src}/**/*`, outputDir);
  });
};

const copyBentleyFrontendAssets = (outputDir: string) => {
  const bentleyPackagesPath = "node_modules/@bentley";
  fs.readdirSync(bentleyPackagesPath).map((packageName) => {
    const packagePath = path.resolve(bentleyPackagesPath, packageName);
    return path.join(packagePath, "lib", "public");
  }).filter((assetsPath) => {
    return fs.existsSync(assetsPath);
  }).forEach((src) => {
    cpx.copySync(`${src}/**/*`, outputDir);
  });
};

class IntegrationTestsApp extends NoRenderApp {
  protected static supplyI18NOptions(): I18NOptions {
    const urlTemplate = "file://" + path.join(path.resolve("lib/public/locales"), "{{lng}}/{{ns}}.json").replace(/\\/g, "/");
    return { urlTemplate };
  }

  public static async startup(opts?: IModelAppOptions): Promise<void> {
    await NoRenderApp.startup({ ...opts, i18n: this.supplyI18NOptions() });
    cpx.copySync(`assets/**/*`, "lib/assets");
    copyBentleyBackendAssets("lib/assets");
    copyBentleyFrontendAssets("lib/public");
  }
}

const initializeCommon = async (props: { backendTimeout?: number, useClientServices?: boolean }) => {
  // init logging
  Logger.initializeToConsole();
  Logger.setLevelDefault(LogLevel.Error);
  Logger.setLevel(LoggingNamespaces.ECObjects_ECExpressions, LogLevel.Warning);
  Logger.setLevel(LoggingNamespaces.ECPresentation, LogLevel.Warning);

  const backendInitProps: PresentationBackendProps = {
    requestTimeout: props.backendTimeout ?? 0,
    rulesetDirectories: ["lib/assets/rulesets"],
    localeDirectories: ["lib/assets/locales"],
    activeLocale: "en-PSEUDO",
    taskAllocationsMap: {
      [RequestPriority.Max]: 1,
    },
    cacheDirectory: path.join("lib", "cache"),
  };
  const frontendInitProps: PresentationFrontendProps = {
    activeLocale: "en-PSEUDO",
  };

  const frontendAppOptions: IModelAppOptions = {
    authorizationClient: props.useClientServices
      ? TestUtility.getAuthorizationClient(TestUsers.regular)
      : undefined,
  };

  const presentationTestingInitProps: PresentationTestingInitProps = {
    backendProps: backendInitProps,
    frontendProps: frontendInitProps,
    frontendApp: IntegrationTestsApp,
    frontendAppOptions,
  };

  await initializeTesting(presentationTestingInitProps);
};

export const initialize = async (backendTimeout: number = 0) => {
  await initializeCommon({ backendTimeout });
};

export const initializeWithClientServices = async () => {
  await initializeCommon({ useClientServices: true });
};

export const terminate = async () => {
  await terminateTesting();
};

export const resetBackend = () => {
  const props = PresentationBackend.initProps;
  PresentationBackend.terminate();
  PresentationBackend.initialize(props);
};
