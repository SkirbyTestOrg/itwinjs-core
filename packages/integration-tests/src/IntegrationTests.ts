/*---------------------------------------------------------------------------------------------
|  $Copyright: (c) 2018 Bentley Systems, Incorporated. All rights reserved. $
 *--------------------------------------------------------------------------------------------*/
// a hack to get imodeljs-frontend import working
(global as any).WebGLRenderingContext = require("gl"); // tslint:disable-line:no-var-requires
// common includes
import TestRpcManager from "@helpers/TestRpcManager";
// backend includes
import { IModelHost } from "@bentley/imodeljs-backend";
import { ECPresentation as ECPresentationBackend, ECPresentation } from "@bentley/ecpresentation-backend";
// frontend includes
import { StandaloneIModelRpcInterface, IModelReadRpcInterface } from "@bentley/imodeljs-common";
import { ECPresentationRpcInterface } from "@bentley/ecpresentation-common";
import { IModelApp } from "@bentley/imodeljs-frontend";
import { ECPresentation as ECPresentationFrontend } from "@bentley/ecpresentation-frontend";

process.env.NODE_ENV = "development";
let isInitialized = false;

export const initialize = () => {
  if (isInitialized)
    return;

  // install mock of browser's XMLHttpRequest for unit tests
  (global as any).XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest; // tslint:disable-line:no-var-requires

  // init backend
  IModelHost.startup();
  ECPresentationBackend.initialize({
    rulesetDirectories: ["assets/rulesets"],
    localeDirectories: ["assets/locales"],
  });

  // set up rpc interfaces
  TestRpcManager.initializeClient([StandaloneIModelRpcInterface, IModelReadRpcInterface, ECPresentationRpcInterface]);

  // init frontend
  IModelApp.startup();
  ECPresentationFrontend.initialize({
    activeLocale: IModelApp.i18n.languageList()[0],
  });

  isInitialized = true;
};

export const terminate = () => {
  if (!isInitialized)
    return;

  // terminate backend
  ECPresentation.terminate();
  IModelHost.shutdown();

  // terminate frontend
  ECPresentation.terminate();
  IModelApp.shutdown();

  isInitialized = false;
};
