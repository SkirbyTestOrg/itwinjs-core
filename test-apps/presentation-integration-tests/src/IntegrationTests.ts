/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
import * as fs from "fs";
import * as path from "path";
import * as cpx from "cpx";
// tslint:disable-next-line:no-direct-imports
import "@bentley/presentation-frontend/lib/test/_helpers/MockFrontendEnvironment";
// common includes
import { I18NOptions } from "@bentley/imodeljs-i18n";
import { Logger, LogLevel } from "@bentley/bentleyjs-core";
import { LoggingNamespaces } from "@bentley/presentation-common";
import { NoRenderApp } from "@bentley/imodeljs-frontend";
import { initialize as initializeTesting, terminate as terminateTesting } from "@bentley/presentation-testing";

process.env.NODE_ENV = "development";

const copyBentleyBackendAssets = (outputDir: string) => {
  const bentleyPackagesPath = "node_modules/@bentley";
  fs.readdirSync(bentleyPackagesPath).map((packageName) => {
    const packagePath = path.resolve(bentleyPackagesPath, packageName);
    return path.join(packagePath, "lib", "assets");
  }).filter((assetsPath) => {
    return fs.existsSync(assetsPath);
  }).forEach((src) => {
    cpx.copySync(`${src}/**/*`, outputDir);
  });
};

const copyBentleyFrontendAssets = (outputDir: string) => {
  const bentleyPackagesPath = "node_modules/@bentley";
  fs.readdirSync(bentleyPackagesPath).map((packageName) => {
    const packagePath = path.resolve(bentleyPackagesPath, packageName);
    return path.join(packagePath, "lib", "public");
  }).filter((assetsPath) => {
    return fs.existsSync(assetsPath);
  }).forEach((src) => {
    cpx.copySync(`${src}/**/*`, outputDir);
  });
};

class IntegrationTestsApp extends NoRenderApp {
  protected static supplyI18NOptions(): I18NOptions {
    const urlTemplate = `file://${path.resolve("lib/public/locales")}/{{lng}}/{{ns}}.json`;
    return { urlTemplate };
  }
  protected static onStartup(): void {
    cpx.copySync(`assets/**/*`, "lib/assets");
    copyBentleyBackendAssets("lib/assets");
    copyBentleyFrontendAssets("lib/public");
  }
}

export const initialize = () => {
  // init logging (enable on demand while debugging)
  Logger.initializeToConsole();
  Logger.setLevel(LoggingNamespaces.ECObjects_ECExpressions, LogLevel.None);
  Logger.setLevel(LoggingNamespaces.ECPresentation, LogLevel.None);

  initializeTesting(
    { rulesetDirectories: ["lib/assets/rulesets"], localeDirectories: ["lib/assets/locales"] },
    IntegrationTestsApp);
};

export const terminate = () => {
  terminateTesting(IntegrationTestsApp);
};
