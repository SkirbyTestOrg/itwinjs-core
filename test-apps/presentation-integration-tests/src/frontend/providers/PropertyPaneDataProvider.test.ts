/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
import { expect } from "chai";
import { initialize, terminate } from "../../IntegrationTests";
import { ModelProps } from "@bentley/imodeljs-common";
import { IModelConnection } from "@bentley/imodeljs-frontend";
import { KeySet } from "@bentley/presentation-common";
import { PresentationPropertyDataProvider } from "@bentley/presentation-components";

describe("PropertyDataProvider", async () => {

  let imodel: IModelConnection;
  let provider: PresentationPropertyDataProvider;
  let physicalModelProps: ModelProps;

  before(async () => {
    initialize();
    const testIModelName: string = "assets/datasets/Properties_60InstancesWithUrl2.ibim";
    imodel = await IModelConnection.openSnapshot(testIModelName);
    physicalModelProps = (await imodel.models.queryProps({ from: "bis.PhysicalModel" }))[0];
    provider = new PresentationPropertyDataProvider(imodel, "SimpleContent");
  });

  after(async () => {
    await imodel.closeSnapshot();
    terminate();
  });

  it("creates empty result when properties requested for 0 instances", async () => {
    provider.keys = new KeySet();
    const properties = await provider.getData();
    expect(properties).to.matchSnapshot();
  });

  it("creates property data when given key with concrete class", async () => {
    provider.keys = new KeySet([physicalModelProps]);
    const properties = await provider.getData();
    expect(properties).to.matchSnapshot();
  });

  it("creates property data when given key with base class", async () => {
    provider.keys = new KeySet([{ className: "BisCore:Element", id: "0x75" }]);
    const properties = await provider.getData();
    expect(properties).to.matchSnapshot();
  });

  it("favorites properties", async () => {
    (provider as any).isFieldFavorite = () => true;
    provider.keys = new KeySet([physicalModelProps]);
    const properties = await provider.getData();
    expect(properties).to.matchSnapshot();
  });

});
