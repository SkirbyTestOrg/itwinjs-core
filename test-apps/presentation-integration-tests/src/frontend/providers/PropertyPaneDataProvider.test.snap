// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyDataProvider creates property data 1`] = `
Object {
  "categories": Array [
    Object {
      "expand": false,
      "label": "Mìsçêllãñéòús",
      "name": "Miscellaneous",
    },
  ],
  "description": "Physical Model",
  "label": "Properties_60InstancesWithUrl2",
  "records": Object {
    "Miscellaneous": Array [
      PropertyRecord {
        "description": "Properties_60InstancesWithUrl2",
        "isMerged": false,
        "isReadonly": true,
        "property": Object {
          "displayLabel": "£ábêl",
          "name": "/DisplayLabel/",
          "typename": "string",
        },
        "value": Object {
          "displayValue": "Properties_60InstancesWithUrl2",
          "value": "Properties_60InstancesWithUrl2",
          "valueFormat": 0,
        },
      },
      PropertyRecord {
        "description": "Properties_60InstancesWithUrl2",
        "isMerged": false,
        "isReadonly": true,
        "property": Object {
          "displayLabel": "Modeled Element",
          "name": "PhysicalModel_ModeledElement",
          "typename": "navigation",
        },
        "value": Object {
          "displayValue": "Properties_60InstancesWithUrl2",
          "value": 28,
          "valueFormat": 0,
        },
      },
    ],
  },
}
`;

exports[`PropertyDataProvider favorites properties 1`] = `
Object {
  "categories": Array [
    Object {
      "expand": true,
      "label": "Fàvórîtë",
      "name": "Favorite",
    },
    Object {
      "expand": false,
      "label": "Mìsçêllãñéòús",
      "name": "Miscellaneous",
    },
  ],
  "description": "Physical Model",
  "label": "Properties_60InstancesWithUrl2",
  "records": Object {
    "Favorite": Array [
      PropertyRecord {
        "description": "Properties_60InstancesWithUrl2",
        "isMerged": false,
        "isReadonly": true,
        "property": Object {
          "displayLabel": "£ábêl",
          "name": "/DisplayLabel/",
          "typename": "string",
        },
        "value": Object {
          "displayValue": "Properties_60InstancesWithUrl2",
          "value": "Properties_60InstancesWithUrl2",
          "valueFormat": 0,
        },
      },
      PropertyRecord {
        "description": "Properties_60InstancesWithUrl2",
        "isMerged": false,
        "isReadonly": true,
        "property": Object {
          "displayLabel": "Modeled Element",
          "name": "PhysicalModel_ModeledElement",
          "typename": "navigation",
        },
        "value": Object {
          "displayValue": "Properties_60InstancesWithUrl2",
          "value": 28,
          "valueFormat": 0,
        },
      },
    ],
    "Miscellaneous": Array [
      PropertyRecord {
        "description": "Properties_60InstancesWithUrl2",
        "isMerged": false,
        "isReadonly": true,
        "property": Object {
          "displayLabel": "£ábêl",
          "name": "/DisplayLabel/",
          "typename": "string",
        },
        "value": Object {
          "displayValue": "Properties_60InstancesWithUrl2",
          "value": "Properties_60InstancesWithUrl2",
          "valueFormat": 0,
        },
      },
      PropertyRecord {
        "description": "Properties_60InstancesWithUrl2",
        "isMerged": false,
        "isReadonly": true,
        "property": Object {
          "displayLabel": "Modeled Element",
          "name": "PhysicalModel_ModeledElement",
          "typename": "navigation",
        },
        "value": Object {
          "displayValue": "Properties_60InstancesWithUrl2",
          "value": 28,
          "valueFormat": 0,
        },
      },
    ],
  },
}
`;
